src/Components/Web.Extensions/src/ProtectedBrowserStorage/ProtectedBrowserStorageServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Components.Web.Extensions;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extension methods for registering Protected Browser Storage services.
+    /// </summary>
+    public static class ProtectedBrowserStorageServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds services for protected browser storage to the specified <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/>.</param>
+        public static void AddProtectedBrowserStorage(this IServiceCollection services)
+        {
+            services.AddScoped<ProtectedLocalStorage>();
```

Comment:
This should chain `services.AddDataProtection()` since it's required for the components to run

Reply:
That does make it a bit less obvious how to configure data protection (`AddDataProtection` returns `IDataProtectionBuilder`), or how to compose it with other DI services that also want to use data protection.

As long as failure to enable data protection produces a clear error at runtime, I think it would be OK to require the developer to call `AddDataProtection` manually. It would show up in all the docs relating to `AddProtectedBrowserStorage`.

Reply:
Users can call `AddDataProtection` separately in their app and it will compose with this. We typically wire up pre-reqs for you:

* https://github.com/dotnet/aspnetcore/blob/87629bbad906e9507026692904b6bcb5021cdd33/src/Middleware/Session/src/SessionServiceCollectionExtensions.cs#L29
* https://github.com/dotnet/aspnetcore/blob/62351067ff4c1401556725b401478e648b66acdc/src/Security/Authentication/Core/src/AuthenticationServiceCollectionExtensions.cs#L24

etc